battenberg/utils.py

```diff
@@ -10,7 +10,11 @@
 
 
 def open_repository(path: str) -> Repository:
-    return Repository(discover_repository(path))
+    repo_path = discover_repository(path)
+    if repo_path:
+        return Repository(repo_path)
+
+    raise ValueError(f'{path} is not a valid repository path.')
```

Comment:
This valueError is probably ignored on line 23-24. 
Could we plan to refactor as open_or_init_repository:
if discover_repository(path): 
repo = Repository(repo_path)
else: 
repo = init_repository(path, initial_head=initial_branch)

Reply:
That's actually intentional behaviour 😅  With the `open_or_init_repository` method we prefer to try to open the path to see if it exists and then create one if it doesn't.

Maybe instead of catching `Exception` I should catch `ValueError` (or better a custom `battenberg.exceptions.OpenRepoError` or similar) then it'd raise on any other conditions which I think is the correct behaviour, will amend!

Reply:
Fixed in https://github.com/zillow/battenberg/pull/23/commits/f77f4a71296fadaca0374e4ee8230ebca0557005